### PR TITLE
docs: add top level introduction

### PR DESCRIPTION
**Pull request by bitgopatmcl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,13 +3,28 @@
 **Disclaimer: This project is currently in an alpha state. Documentation is actively
 being worked on.**
 
+A collection of packages for defining and using type-checked apis with TypeScript.
+
+TypeScript is a very powerful type system layered on top of JavaScript, but one drawback
+it has is that this extra type information is not available at runtime.
+[io-ts](https://github.com/gcanti/io-ts) fills in this functionality gap.
+`io-ts-http`(packages/io-ts-http/README.md) builds on top of `io-ts` to make it possible
```
Comment (ericcrosson-bitgo) [inline README.md:11]:
```suggestion
[`io-ts-http`](packages/io-ts-http/README.md) builds on top of `io-ts` to make it possible
```

Seems helpful to provide a hyperlink here

**Review by ericcrosson-bitgo — APPROVED** [review]
I dig it!

One general principle I think we can improve on is "use fewer words". That was my big take-away from working with Gwen to improve my technical writing

For example, instead of 

> TypeScript is a very powerful tyep system layered on top of JavaScript, but one drawback it has is that this extra type information is not available at runtime

we could write

> TypeScript is a very powerful type system layered on top of JavaScript, but lacks type information available at runtime

replacing 8 words with 1!

I feel we could even improve the first part of this sentence (before the comma), but iteration is the name of the game.

Thanks for putting this documentation out! :rocket:

File: README.md [hunk 2]
```diff
@@ -3,13 +3,28 @@
 **Disclaimer: This project is currently in an alpha state. Documentation is actively
 being worked on.**
 
+A collection of packages for defining and using type-checked apis with TypeScript.
+
+TypeScript is a very powerful type system layered on top of JavaScript, but one drawback
+it has is that this extra type information is not available at runtime.
```
[the remark this answers is not in the document]
Comment (bitgopatmcl) [reply README.md:9]: Good idea

**Review by ericcrosson-bitgo — APPROVED** [review]
:man_dancing: noice!

**Pull request merged**